File: runtime/app/schemas/record.py
"""Tyndale Record API shapes (D5, Phase C)."""

from __future__ import annotations

from pydantic import BaseModel


class DeadlineInfo(BaseModel):
    label: str
    due_date: str | None = None
    source: str  # provenance (appeals rule source, or the deadline_type)


class ThreeNumberBrief(BaseModel):
    provider_billed: float
    eob_member_responsibility: float
    tyndale_computed: float


class SubCaseRow(BaseModel):
    case_file_id: str
    service_date: str | None = None  # best-effort; null until a structured visit date is extracted
    provider: str | None = None
    status: str
    label: str
    resume: str  # 'summary' (results-bearing) | 'thread' (in-flight)
    three_number: ThreeNumberBrief | None = None  # null → the row shows needs-documents, not {0,0,0}
    open_item_count: int = 0
    next_deadline: DeadlineInfo | None = None
    recovered_so_far: float = 0.0  # CONFIRMED only


class RecordAggregates(BaseModel):
    total_billed_reviewed: float
    total_recovered: float  # CONFIRMED outcomes only, shown "so far"
    total_identified: float  # audit ESTIMATE (finding gap) — labeled separately, never recovered
    open_items: int
    next_check_in_date: str | None = None


class RecordPayload(BaseModel):
    window_months: int
    sub_cases: list[SubCaseRow]
    aggregates: RecordAggregates
    has_older: bool = False  # older-than-window cases exist → "full history"

File: packages/shared/src/record.ts
// Tyndale Record API shapes (D5, Phase C — DL-91). Mirrors runtime/app/schemas/record.py.

export interface DeadlineInfo {
  label: string;
  due_date: string | null;
  source: string;
}

export interface ThreeNumberBrief {
  provider_billed: number;
  eob_member_responsibility: number;
  tyndale_computed: number;
}

export interface SubCaseRow {
  case_file_id: string;
  service_date: string | null;
  provider: string | null;
  status: string;
  label: string;
  resume: 'summary' | 'thread';
  three_number: ThreeNumberBrief | null; // null → the row shows needs-documents, not {0,0,0}
  open_item_count: number;
  next_deadline: DeadlineInfo | null;
  recovered_so_far: number; // CONFIRMED only
}

export interface RecordAggregates {
  total_billed_reviewed: number;
  total_recovered: number; // CONFIRMED outcomes only, "so far"
  total_identified: number; // audit ESTIMATE — labeled separately, never recovered
  open_items: number;
  next_check_in_date: string | null;
}

export interface RecordPayload {
  window_months: number;
  sub_cases: SubCaseRow[];
  aggregates: RecordAggregates;
  has_older: boolean;
}
